src/main/java/core/mvc/DispatcherServlet.java

```diff
@@ -38,25 +40,38 @@ public void init() {
         handlerAdapterRegistry.addHandlerAdapter(new ControllerHandlerAdapter());
 
         handlerExecutor = new HandlerExecutor(handlerAdapterRegistry);
+
+        handlerInterceptorRegistry = new HandlerInterceptorRegistry();
+        handlerInterceptorRegistry.addInterceptor(new ExecutionTimeLogInterceptor());
     }
 
     @Override
-    protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String requestUri = req.getRequestURI();
-        logger.debug("Method : {}, Request URI : {}", req.getMethod(), requestUri);
+    protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String requestUri = request.getRequestURI();
+        logger.debug("Method : {}, Request URI : {}", request.getMethod(), requestUri);
+
+        Object handler = null;
 
         try {
-            Optional<Object> maybeHandler = handlerMappingRegistry.getHandler(req);
-            if (!maybeHandler.isPresent()) {
-                resp.setStatus(HttpStatus.NOT_FOUND.value());
+            Optional<Object> maybeHandler = handlerMappingRegistry.getHandler(request);
+            if (maybeHandler.isEmpty()) {
+                response.setStatus(HttpStatus.NOT_FOUND.value());
                 return;
             }
 
+            handler = maybeHandler.get();
+
+            if (handlerInterceptorRegistry.applyPreHandle(request, response, handler)) {
```

Comment:
```suggestion
            if (!handlerInterceptorRegistry.applyPreHandle(request, response, handler)) {
```
false 값이 리턴되는 경우 실행을 멈추어야 하지 않을까요 ?

Reply:
앗 이런 실수를... 😢 꼼꼼하게 봐주셔서 감사합니다 🙏